lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -107,14 +107,25 @@ impl<TInvertedIndex: InvertedIndex> SparseVectorIndex<TInvertedIndex> {
         let mut index_point_count: usize = 0;
         for id in borrowed_id_tracker.iter_ids_excluding(deleted_bitslice) {
             check_process_stopped(stopped)?;
-            let vector = borrowed_vector_storage.get_vector(id);
-            let vector: &SparseVector = vector.as_vec_ref().try_into()?;
-            // do not index empty vectors
-            if vector.is_empty() {
-                continue;
+            // It is possible that the vector is not present in the storage in case of crash.
+            // Because:
+            // - the `id_tracker` is flushed before the `vector_storage`
+            // - the sparse index *before* recovering the WAL when loading a segment
+            match borrowed_vector_storage.get_vector_opt(id) {
```

Comment:
We know exactly that here we have `SparseVectorStorage` type as vector storage. We can use `get_sparse` and remove `get_vector_opt`at all

Reply:
Do you want to pattern match the storage right there instead?

Reply:
as if get_sparse return optional

Reply:
I prefer this one better